src/Interactive/Features/Interactive/Core/InteractiveHost.LazyRemoteService.cs

```diff
@@ -62,20 +62,23 @@ public void Dispose(bool joinThreads)
                     }
 
                     bool initializing = true;
-                    cancellationToken.Register(() =>
-                    {
-                        if (initializing)
+                    RemoteExecutionResult initializationResult;
+                    using (cancellationToken.Register(
+                        () =>
```

Comment:
I am not sure about the intent, but it seems you are deregistering the callback earlier now.

---
In reply to: [121204270](https://github.com/dotnet/roslyn/pull/20111#discussion_r121204270) [](ancestors = 121204270)